lib/sql-string.js

```diff
@@ -193,14 +193,16 @@ SqlString.bufferToString = function(buffer, dialect) {
 };
 
 SqlString.objectToValues = function(object, timeZone) {
-  var values = [];
-  for (var key in object) {
```

Comment:
I agree, many of these changes make the code better :)

Performance is always important, but not for the sake of readability unless it's very hot code (which you haven't really argued for yet).

`while pop` looks good, but it'd still like to see what the difference would be between `for`, `while pop` and lodash. If the improvement is minuscule i prefer lodash.

Can you do a quick benchmark on the current implementation, lodash and:

```js
var keys = Object.keys (something)
    , key;
while (key = keys.pop()) {}
```